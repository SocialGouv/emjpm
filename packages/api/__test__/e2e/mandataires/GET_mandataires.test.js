//

const request = require("supertest");

const { knex } = global;
jest.setMock("@emjpm/api/db/knex", knex);

const server = require("@emjpm/api/app");

const { getTokenByUserType } = require("../utils");

//

beforeAll(async () => {
  await knex.migrate.latest();
  await knex.seed.run();
});

// strip the created_at date in the response
// eslint-disable-next-line no-unused-vars
const simpler = ({ created_at, date_mesure_update, ...props }) => props;

test("should GET mandataire profile", async () => {
  const token = await getTokenByUserType("mandataire");
  const response = await request(server)
    .get("/api/v1/mandataires/1")
    .set("Authorization", "Bearer " + token);

  expect(simpler(response.body)).toMatchSnapshot();

  expect(response.status).toBe(200);
});

test("TI should NOT GET mandataire profile", async () => {
  const token = await getTokenByUserType("ti");
  const response = await request(server)
    .get("/api/v1/mandataires/1")
    .set("Authorization", "Bearer " + token);

  expect(simpler(response.body)).toMatchSnapshot({
    stack: expect.any(String)
  });

  expect(response.status).toBe(401);
});

test("TI should GET list of mandataires", async () => {
  const token = await getTokenByUserType("ti");
  const response = await request(server)
    .get("/api/v1/mandataires")
    .set("Authorization", "Bearer " + token);

  expect(response.body.map(simpler)).toMatchSnapshot();

  expect(response.status).toBe(200);
});
