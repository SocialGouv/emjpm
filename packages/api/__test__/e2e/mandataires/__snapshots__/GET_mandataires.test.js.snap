// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TI should GET list of mandataires 1`] = `
Array [
  Object {
    "adresse": "21 rue de houx",
    "code_postal": "62000",
    "date_mesure_update": null,
    "dispo_max": 3,
    "email": "ud@ud.com",
    "email_send": null,
    "etablissement": "UDAHF",
    "genre": null,
    "id": 1,
    "latitude": 50.5333,
    "longitude": 2.6333,
    "mesures_en_attente": null,
    "mesures_en_cours": 2,
    "nb_secretariat": 2,
    "nom": null,
    "prenom": "Adrien",
    "secretariat": false,
    "telephone": "0237100000",
    "telephone_portable": "0101010101",
    "type": "individuel",
    "user_id": 1,
    "ville": "Arras",
    "zip": null,
  },
  Object {
    "adresse": "21 rue monge",
    "code_postal": "62000",
    "date_mesure_update": null,
    "dispo_max": 3,
    "email": "panam@paris.com",
    "email_send": null,
    "etablissement": "test42",
    "genre": null,
    "id": 2,
    "latitude": 50.5333,
    "longitude": 2.6333,
    "mesures_en_attente": null,
    "mesures_en_cours": 2,
    "nb_secretariat": 2,
    "nom": null,
    "prenom": "Julien",
    "secretariat": true,
    "telephone": "0237100000",
    "telephone_portable": "0101010101",
    "type": "prepose",
    "user_id": 2,
    "ville": "Paris",
    "zip": null,
  },
]
`;

exports[`TI should NOT GET mandataire profile 1`] = `
Object {
  "message": "Please log in",
  "name": "UnauthorizedError",
  "stack": Any<String>,
  "status": 401,
}
`;

exports[`should GET mandataire profile 1`] = `
Object {
  "adresse": "21 rue de houx",
  "code_postal": "62000",
  "date_mesure_update": null,
  "dispo_max": 3,
  "email": "ud@ud.com",
  "email_send": null,
  "etablissement": "UDAHF",
  "genre": null,
  "id": 1,
  "latitude": 50.5333,
  "longitude": 2.6333,
  "mesures_en_attente": null,
  "mesures_en_cours": 2,
  "nb_secretariat": 2,
  "nom": null,
  "prenom": "Adrien",
  "secretariat": false,
  "telephone": "0237100000",
  "telephone_portable": "0101010101",
  "type": "individuel",
  "user_id": 1,
  "ville": "Arras",
  "zip": null,
}
`;
